tools/walletextension/wallet_extension.go

```diff
@@ -119,17 +120,31 @@ func (we *WalletExtension) handleHTTPEthJSON(resp http.ResponseWriter, req *http
 	method := reqJSONMap[reqJSONKeyMethod]
 	fmt.Printf("Received request from wallet: %s\n", body)
 
-	//// We encrypt the JSON with the enclave's public key.
-	//fmt.Println("🔒 Encrypting request from wallet with enclave public key.")
-	//eciesPublicKey := ecies.ImportECDSAPublic(we.enclavePublicKey)
-	//encryptedBody, err := ecies.Encrypt(rand.Reader, eciesPublicKey, body, nil, nil)
-	//if err != nil {
-	//	logAndSendErr(resp, fmt.Sprintf("could not encrypt request with enclave public key: %s", err))
-	//	return
-	//}
+	// We encrypt the request's params with the enclave's public key if it's a sensitive request.
+	maybeEncryptedBody := body
+	if isSensitive(method) {
+		fmt.Println("🔒 Encrypting request from wallet with enclave public key.")
+		params := reqJSONMap[reqJSONKeyParams]
+		paramsJSON, err := json.Marshal(params)
```

Comment:
Again there's a lot of low-level fiddling here and the method becomes quite hard to follow (getting towards 100 lines long).

Maybe we could break it up into some high-level chunks that hide away the details for readability, like something like this could be easier to scan for the reader maybe:
```
fwdReq, err = encryptRequestIfNeeded(req.body)

resp, err = forwardRequestAndAwaitResponse(fwdReq)

clientResp, err = decryptResponseIfNeeded(resp)

returnResponse(clientResp)
```

Reply:
I agree but I didn't want to start doing refactoring here. Will do in a follow-up?